design/ceph/object/store.md

```diff
@@ -70,6 +72,11 @@ If `preservePoolsOnDelete` is set to 'true' the pools used to support the object
     erasureCoded:
       dataChunks: 6
       codingChunks: 2
+  radosNamespaces:
```

Comment:
I think the design could be simplified by doing something like below... 
```suggestion
  metadataPool:
    useSharedPool:
      name: rgw-meta-pool
    failureDomain: host
    replicated:
      size: 3
  dataPool:
    useSharedPool:
      name: rgw-data-pool
    failureDomain: device
    erasureCoded:
      dataChunks: 6
      codingChunks: 2
```

... however, there is still a problem if pool configs are set by multiple object stores. While it is clunky, I don't see a way to resolve this without requiring a CephBlockPool to be created for RGWs before they use the pool. Multiple CephObjectStore resources shouldn't be contending over ownership of the pool definitions.

Reply:
Also, asking users to create pre-requisite pools for their object stores that are going to share pool access doesn't seem like it changes anything from our existing usage patterns for things like the `.nfs` pool.

```
  metadataPool:
    useSharedPool: # if present, no other configs are allowed
      name: rgw-meta-pool
    failureDomain: host # error because not allowed
    replicated:
      size: 3
  dataPool:
    useSharedPool: # ditto
      name: rgw-data-pool
    failureDomain: device # ditto
    erasureCoded:
      dataChunks: 6
      codingChunks: 2
```

Reply:
I like the idea of specifying the pools under the new radosNamespaces section, with a couple suggestions:
- The shared pool name always needs to be specified, whether the pools are defined in another CR or not.
- Only one object CR is expected to define the pools. If multiple CRs define the same exact pools, then there is no problem. If the pool definitions conflict, the admission controller could check, although it's not always enabled. But honestly this conflict would be an annoying design issue.

The spec of the CR that owns the pools:
```yaml
  radosNamespaces:
    metadataPool:
      sharedPoolName: rgw-meta-pool
      failureDomain: host
      replicated:
        size: 3
    dataPool:
      sharedPoolName: rgw-data-pool
      failureDomain: osd
      erasureCoded:
        dataChunks: 6
        codingChunks: 2
```

Spec of the other CRs:
```yaml
  radosNamespaces:
    metadataPool:
      sharedPoolName: rgw-meta-pool
    dataPool:
      sharedPoolName: rgw-data-pool
```

> Also, asking users to create pre-requisite pools for their object stores that are going to share pool access doesn't seem like it changes anything from our existing usage patterns for things like the .nfs pool.

True, we did something special for .nfs with the CephBlockPool CR, so if we were to do something special for the RGW pools with CephBlockPool, maybe it wouldn't be so bad as I was thinking. There are just a couple behaviors specific to RGW pools, such as setting a lower PG count for metadata pools and setting the application to "rgw" instead of "rbd". So perhaps it would then look like this... I'd like to avoid exposing the `application` in the pool spec, not sure if the default `rbd` application would be an issue from the CephBlockPool.

Metadata CephBlockPool:
```yaml
  metadata:
    name: rgw-meta-pool
  spec:
    failureDomain: host
    replicated:
      size: 3
    parameters:
      pg_num: 8
    application: rgw
```

Data CephBlockPool:
```yaml
  metadata:
    name: rgw-data-pool
  spec:
      failureDomain: osd
      erasureCoded:
        dataChunks: 6
        codingChunks: 2
    application: rgw
```

CephObjectStore CR:
```yaml
  radosNamespaces:
    metadataPoolName: rgw-meta-pool
    dataPoolName: rgw-data-pool
```